include/cantera/thermo/Elements.h

```diff
@@ -128,7 +128,7 @@ double getElementWeight(const std::string& ename);
  * @exception CanteraError if the element has no stable isotopes, and
  * therefore no standard atomic weight
  */
-double getElementWeight(int atomicNumber);
+double getElementWeight(size_t atomicNumber);
```

Comment:
I had intentionally left these as `int`, as semantically the atomic number is not a size (of an array). I guess that's a pretty pedantic point, but is there another way to resolve this warning that maintains the semantics of the types?

Reply:
I see your intent, but subsequently `atomicNumber` is used to pick elements out of the array `atomicWeightTable` :smile: ...

Reply:
I'm not sure how much to push back, I'm not sure how much the semantics here really matter... I don't think either of us feels very strongly about this change but it just feels a little tiny bit... odd... to make this change. The impact to users is effectively nil, so this is me being purely academic and pedantic because Cantera is the project where I can do that 😜 

That said, Boost has a function to safely convert unsigned to signed integer, [here](https://www.boost.org/doc/libs/1_55_0/libs/numeric/conversion/doc/html/boost_numericconversion/improved_numeric_cast__.html#boost_numericconversion.improved_numeric_cast__.numeric_cast) (I didn't check if it was header-only). So casting `size_t` to `int` is at least feasible 🤷‍♂️

Another option, since atomic numbers can't physically be negative, is to do a local `typedef` for an unsigned integer (it could even be an 8 bit integer, since there are only a few hundred elements). I suspect that kind of automatic type casting won't result in warnings from the compiler?

On the other hand, an unsigned integer doesn't actually enforce that the number is positive, either at compile time or runtime, so in that sense it's also not what we want this type to really represent either.

Clearly, I've thought about this _way_ more than it needs to be. In the end, I'm fine with having said everything that's above so that it's out of my head and other people can judge how crazy it is, and I'm fine with whatever decision we think is best for Cantera maintenance. Hence, this is a comment instead of requested change ☺️

Reply:
The [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html#Integer_Types), which we reference but don't strictly follow, recommends against using the unsigned types in most cases, and offers some rationale as to why.

The fairly consistent use of `size_t` was actually something I'd introduced before the release of Cantera 2.0, but I doubt I'd make the same decision now. While the standard library uses `size_t` extensively, some libraries that we work with like SUNDIALS and Eigen use signed integers for sizes.

I'm not suggesting transitioning back to signed integers broadly, but in this particular instance, I think the additional rationale for wanting to use `int` makes it a reasonable choice.

Reply:
Interesting. It is correct that I don’t feel strongly either way - the switch to `size_t` was just the most efficient way to get rid of the warnings. Ultimately, this PR is trivial and not high priority.

Reply:
> While the standard library uses size_t extensively

FWIW, this is why I changed the return types of the `num*` functions when I updated the overall API. I still think these warnings are worth resolving, though... Especially because this PR addresses more than just the elements case